crates/net/downloaders/src/bodies/task.rs

```diff
@@ -115,35 +115,23 @@ impl<T: BodyDownloader> Future for SpawnedDownloader<T> {
         let this = self.get_mut();
 
         loop {
-            loop {
-                match this.updates.poll_next_unpin(cx) {
-                    Poll::Pending => break,
-                    Poll::Ready(None) => {
-                        // channel closed, this means [TaskDownloader] was dropped, so we can also
-                        // exit
-                        return Poll::Ready(())
-                    }
-                    Poll::Ready(Some(range)) => {
-                        if let Err(err) = this.downloader.set_download_range(range) {
-                            tracing::error!(target: "downloaders::bodies", ?err, "Failed to set bodies download range");
-
-                            match ready!(this.bodies_tx.poll_reserve(cx)) {
-                                Ok(()) => {
-                                    if this.bodies_tx.send_item(Err(err)).is_err() {
-                                        // channel closed, this means [TaskDownloader] was dropped,
-                                        // so we can also
-                                        // exit
-                                        return Poll::Ready(())
-                                    }
-                                }
-                                Err(_) => {
-                                    // channel closed, this means [TaskDownloader] was dropped, so
-                                    // we can also exit
-                                    return Poll::Ready(())
-                                }
-                            }
+            while let Poll::Ready(update) = this.updates.poll_next_unpin(cx) {
+                if let Some(range) = update {
+                    if let Err(err) = this.downloader.set_download_range(range) {
+                        tracing::error!(target: "downloaders::bodies", ?err, "Failed to set bodies download range");
+
+                        let forward_error_result = ready!(this.bodies_tx.poll_reserve(cx))
```

Comment:
even though this should not really be reachable, this still throws away the error if the channel is currently full.

we could clone the `bodies_tx` first which would ensure a new slot for the error